### PR TITLE
inventory_ui: improve pickup header

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary
None

#### Purpose of change
The instructions in the pickup header blend in and are too wordy to read so new players don't realize there are helpful keys available. The header also gets garbled on the smallest screens.

#### Describe the solution
Screenshot from 2023-03-29 12-44-11

<details>
<summary>Before, for reference</summary>

Screenshot from 2023-03-29 10-49-58

</details>

Also update the default keybindings to make this work. Namely, add a key for `Mark a specific amount of selected items` and remove the double keybindings from `e`xamine as `E` has been repurposed in #62864.

Some internal changes were needed to preserve numpad navigation mode, see https://github.com/CleverRaven/Cataclysm-DDA/pull/64664#issuecomment-1488371822.

#### Describe alternatives you've considered
N/A

#### Testing
Other inventory UIs aren't affected

#### Additional context
N/A

### CODE REVIEW

**Comment by mqrause** [comment]
> add a key for `Mark a specific amount of selected items`

Doesn't this change behavior because you can't just type a number anymore?

**Comment by andrei8l** [comment]
> > add a key for `Mark a specific amount of selected items`
> 
> Doesn't this change behavior because you can't just type a number anymore?

Weird. Why is that a thing? I can remove that check.

**Comment by andrei8l** [comment]
I've removed the exclusion so you can now both `Mark a specific amount...` and type a number then toggle. I can't find any ill effects and it was probably a very old hack. I've also re-added a bit of text about typing number.

**Comment by andrei8l** [comment]
Alright, I've implemented [Kevin's  suggestion](https://github.com/CleverRaven/Cataclysm-DDA/pull/51085#issuecomment-907557439). Namely:
1. There's a `TOGGLE_NUMPAD_NAVIGATION` key and associated `uistate` that defaults to false/off
2. If numpad navigation is off, typing a number opens the string input bar so you can see what you're typing. You can close the string input bar in this case with `TOGGLE_ENTRY` so there is no functional difference
3. If numpad navigation is on, the behaviour from 2 is disabled but you can still use `MARK_WITH_COUNT`
4. `MARK_WITH_COUNT` is always available and has the current behaviour

There's also an extra hint on the right side when numpad navigation is active
Screenshot from 2023-03-29 13-28-12
